int32 pointX
int32 pointY
